UTRannotator.pm

```diff
@@ -0,0 +1,1351 @@
+      $uAUG_gained_type = "uORF";
+    }
+    else{
+      if(($mut_utr_length - $pos_A) % 3){
+        $uAUG_gained_type = "OutOfFrame_oORF";
+      }
+      else{
+        $uAUG_gained_type = "inFrame_oORF";
+      }
+    }
+
+    my @overlapping_seq = split //, $mut_utr_seq.$UTR_info->{cds_seq};
+    my %existing_uORF = %{$self->existing_uORF(\@overlapping_seq)};
+    if(exists($existing_uORF{$pos_A})){
+      my @stop_pos_array = sort{$a<=>$b}@{$existing_uORF{$pos_A}};
+      my $stop_pos = $stop_pos_array[0];
+      $uAUG_gained_DistanceToStop = $stop_pos-$pos_A;
+    } else {
+      $uAUG_gained_DistanceToStop = "NA";
+    }
+
+
+    my %uORF_effect = (
+            "uAUG_gained_KozakContext" => $uAUG_gained_KozakContext,
+            "uAUG_gained_KozakStrength" => $uAUG_gained_KozakStrength,
+            "uAUG_gained_DistanceToCDS" => $uAUG_gained_DistanceToCDS,
+            "uAUG_gained_type" => $uAUG_gained_type,
+            "uAUG_gained_DistanceToStop" => $uAUG_gained_DistanceToStop,
+            "uAUG_gained_CapDistanceToStart" => $uAUG_gained_DistanceFromCap,
+    );
+
+    $output_flag = "uAUG_gained";
+    $output_effects = $self->transform_hash_to_string(\%uORF_effect);
+  }
+
+  $result{'uAUG_gained_flag'} = $output_flag;
+  $result{'uAUG_gained_effect'} = $output_effects;
+
+  return \%result;
+}
+
+sub uSTOP_gained {
+  # Description: annotate whether a five_prime_UTR_variant creates new stop codon. It only evaluate SNVs.
+
+  my ($self, $variant_info,$UTR_info, %kozak_strength) = @_;
+
+  my $chr = $variant_info->{chr};
+  my $pos = $variant_info->{pos};
+  my $ref = $variant_info->{ref};
+  my $alt = $variant_info->{alt};
+
+  my @sequence = split //, $UTR_info->{seq};
+  my $strand = $UTR_info->{strand};
+  my $utr_length = @sequence;
+
+	#return annotators
+	my $uSTOP_gained_ref_StartDistanceToCDS = "";  # the distance between the uAUG of the disrupting uORF to CDS
+	my $uSTOP_gained_KozakContext = "";  # the Kozak context sequence of the disrupting uORF
+	my $uSTOP_gained_KozakStrength = ""; # the Kozak strength of the the disrupting uORF
+	my $uSTOP_gained_ref_type = ""; # the type of uORF being disrupted - any of the following: uORF, inframe_oORF,OutOfFrame_oORF
+	my $uSTOP_gained_newSTOPDistanceToCDS = ""; # the distance between the gained uSTOP to the start of the CDS
+    my $uSTOP_gained_evidence = ""; # the translation evidence of the disrupted uORF
+
+	#indicate whether the variant creates a stop codon
+	my $flag = 0;
+	my $current_kozak = "";
+	my $current_kozak_strength ="";
+
+	#the relative position of input variant in the UTR sequence
+
+    my %result = (); #result is a hash table with two elements: $flag and $output_effects
+    my $output_flag = 0;
+    my $output_effects = "";
+
+    my %existing_ref_uORF = %{$self->existing_uORF(\@sequence)};
+
+    my $ref_coding = $self->get_ref_coding($ref);
+    my $alt_coding = $alt eq "-"? "" : $alt;
+    reverse_comp(\$alt_coding) if($strand < 0);
```

Comment:
You could replace it with:
`my $alt_coding = $self->get_alt_coding($alt,$strand);`